packages/cli/src/services/import.service.ts

```diff
@@ -57,10 +58,14 @@ export class ImportService {
 
 				const workflowId = upsertResult.identifiers.at(0)?.id as string;
 
-				await tx.upsert(SharedWorkflow, { workflowId, userId, role: 'workflow:owner' }, [
-					'workflowId',
-					'userId',
-				]);
+				const personalProject =
+					await Container.get(ProjectRepository).getPersonalProjectForUserOrFail(userId);
```

Comment:
I'm not sure whether it makes sense to use the same transaction as context for this query, since we're inside a transaction already anyway. In any case, I don't exactly know how we could even achieve that, it's just an idea.

Reply:
I'll merge it since it's approved and necessary for the decorator PR.

I'll create a followup PR trying what you suggested.

Reply:
For the record: We've decided not to change this, as passing the transaction/entitymanager around as an optional argument is noisy and hoping that every service and repo uses it correctly is error prone. There are also considerations around how sqlite implements transactions and how they affect connection pooling in postgres. 
So for now this stays as is.